src/main/java/uk/gov/hmcts/sptribs/caseworker/event/ReinstateCase.java

```diff
@@ -78,22 +80,46 @@ public SubmittedCallbackResponse reinstated(CaseDetails<CaseData, State> details
         var cicCase = details.getData().getCicCase();
         final StringBuilder messageLine2 = new StringBuilder(100);
         messageLine2.append(" A notification will be sent via email to: ");
-        if (!CollectionUtils.isEmpty(cicCase.getNotifyPartySubject())) {
+
+        if (!CollectionUtils.isEmpty(cicCase.getNotifyPartySubject())
+            && StringUtils.hasText(cicCase.getEmail())) {
             messageLine2.append("Subject, ");
             cicCase.setNotifyPartySubject(null);
         }
         if (!CollectionUtils.isEmpty(cicCase.getNotifyPartyRespondent())) {
             messageLine2.append("Respondent, ");
             cicCase.setNotifyPartyRespondent(null);
         }
-        if (!CollectionUtils.isEmpty(cicCase.getNotifyPartyRepresentative())) {
+        if (!CollectionUtils.isEmpty(cicCase.getNotifyPartyRepresentative())
+            && StringUtils.hasText(cicCase.getRepresentativeEmailAddress())) {
             messageLine2.append("Representative, ");
             cicCase.setNotifyPartyRepresentative(null);
         }
```

Comment:
Just a suggestion: Code might be more readable if we extract the code related to post and email into 2 private methods.